src/main/java/com/geronimo/service/IMessageService.java

```diff
@@ -1,16 +1,16 @@
 package com.geronimo.service;
 
+import com.geronimo.exceptions.RebloggingOwnMessageException;
 import com.geronimo.model.Message;
 import com.geronimo.model.User;
-import org.springframework.transaction.annotation.Transactional;
 
 public interface IMessageService {
 
     void saveOrUpdateMessage(Message message);
 
     void postMessage(Message message);
 
-    void reblogMessage(Message messageToReblog, User whoReblogged);
+    void reblogMessage(Message messageToReblog, User whoReblogged) throws RebloggingOwnMessageException;
```

Comment:
As we have RebloggingOwnMessageExeption being a RuntimeException we don't need throws